index.bs

```diff
@@ -258,6 +262,67 @@ occurred on the [=remote end=].
     - <code>params</code> which defines a mapping containing event data. The
       populated value of this map is the <dfn export for=command>event
       parameters</code>.
+ - A set of [=remote end steps=] which define how to construct the [=event
+   type=] data.
+ - An <dfn export for=event>event trigger</dfn> which is a condition on which
```

Comment:
Well I don't see how we can make it procedural unless we actually patch every other spec to have hooks that call into WebDriver (aiui that's basically what happens with CDP). For example we want to fire an event when a page is loaded. So we need somewhere to write "when an event named 'load' is fired at a global object".

Reply:
We don't need other specs to invoke WebDriver BiDi algorithms, just have the appropriate definitions that we can hook into. For the load event example, it might be https://html.spec.whatwg.org/#completely-finish-loading

Whether we have an explicit "event trigger" concept or not, we'll need the same from other specs. I just don't quite see why we need the concept, when we can say things like "When a document completely finishes loading: do our stuff" instead of defining the trigger and some steps separately.

I might be wrong, but if we could leave out "event trigger" until we define an event, it'll be much easier to judge.

Reply:
OK, I've merged these concepts, but I still think this should be called a "remote end event trigger" or something, since it's not a set of steps that get directly invoked by this spec but a COMEFROM for hooks in other specs.

Reply:
Alright, we'll just see how it turns out, given two or three real cases it should be clear which style works best.